src/ResourceManagement/ResourceManager/Domain/PolicyDefinition/Definition/IDefinition.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Management.ResourceManager.Fluent.PolicyDefinition.Definition
+{
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Models;
+    using System;
+
+    /// <summary>
+    /// The entirety of the PolicyDefinition definition.
+    /// </summary>
+    public interface IDefinition :
+        IBlank,
+        IWithCreate
+    {
+    }
+
+    /// <summary>
+    /// The first stage of a PolicyDefinition definition.
+    /// </summary>
+    public interface IBlank :
+        IWithPolicyRule
+    {
+    }
+
+    /// <summary>
+    /// The stage of the PolicyDefinition definition allowing to specify policy rule.
+    /// </summary>
+    public interface IWithPolicyRule
+    {
+        /// <summary>
+        /// Specifies the policy rule of PolicyDefinition.
+        /// </summary>
+        /// <param name="policyRule">The policy rule of the PolicyDefinition.</param>
+        /// <return>The next stage of the definition.</return>
+        IWithCreate WithPolicyRule(Object policyRule);
```

Comment:
Any idea why we would need to support a Object as input?

Reply:
This is how it gets defined in [definition swagger](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Authorization/stable/2019-06-01/policyDefinitions.json#L438)

Reply:
Do you know what object this is? If you do not know, do you think SDK user would know?

Reply:
Good catch. We could ask this question to [service side](https://docs.microsoft.com/en-us/rest/api/resources/policydefinitions/createorupdate)

Reply:
What the value from the Java side?

Reply:
In portal, the policy rule is json format. So it makes sense to have object parameters here. We have implemented object and json string both java and .net sides.

Reply:
Please document this find on the API docs.